python/cudf/cudf/core/frame.py

```diff
@@ -2688,7 +2688,13 @@ def acos(self):
                     1.5707963267948966,  1.266103672779499],
                     dtype='float64')
         """
-        return self._unaryop("acos")
+        if self.iloc[0].dtype == "int":
+            data = self.astype("float64")
+            data = data._unaryop("acos")
+        else:
+            data = self._unaryop("acos")
+        newdata = data.mask(data < 0)
```

Comment:
Should we also mask data that is beyond `pi`+1 ? The reason we are masking here is probably because of garbage values(looks like int min values).

Reply:
It looked like the garbage values were all below 0. Happy to try to add that condition if you think it would help

Reply:
I feel it'd be safe to have both upper and lower bounds.

Reply:
Makes sense, I'll update this!

Reply:
Added the upper bound as well :D